Userland/Applications/Presenter/SlideObject.cpp

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2022, kleines Filmröllchen <filmroellchen@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "SlideObject.h"
+#include <AK/JsonObject.h>
+#include <AK/RefPtr.h>
+#include <LibCore/Object.h>
+#include <LibCore/Stream.h>
+#include <LibGUI/Margins.h>
+#include <LibGfx/Font/FontDatabase.h>
+#include <LibGfx/Forward.h>
+#include <LibGfx/Orientation.h>
+#include <LibGfx/Painter.h>
+#include <LibGfx/Size.h>
+#include <LibGfx/TextWrapping.h>
+#include <LibImageDecoderClient/Client.h>
+
+ErrorOr<NonnullRefPtr<SlideObject>> SlideObject::parse_slide_object(JsonObject const& slide_object_json, NonnullRefPtr<GUI::Window> window)
+{
+    auto image_decoder_client = TRY(ImageDecoderClient::Client::try_create());
```

Comment:
Why do we need a new image decoder client for every slide? Can't we re-use one at the application level? 

Only image objects need it as well, so we create this even if we load text